### PR TITLE
Extend Security configuration to assign roles to namespaces

### PR DESCRIPTION
**Pull request by uweschaefer**

### CODE REVIEW

**Comment by uweschaefer** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 2
           

Complexity increasing per file
==============================
- factcast-server-grpc/src/main/java/org/factcast/server/grpc/FactCastSecurityConfiguration.java  1
- factcast-server-grpc/src/main/java/org/factcast/server/grpc/auth/FactCastAccessConfiguration.java  1
- factcast-server-grpc/src/main/java/org/factcast/server/grpc/auth/FactCastAccount.java  3
- factcast-server-grpc/src/main/java/org/factcast/server/grpc/auth/FactCastUser.java  1
- factcast-server-grpc/src/main/java/org/factcast/server/grpc/auth/AccessRules.java  5
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/uwe/factcast/pullRequest?prid=5054049&bid=16638135)

File: factcast-server-grpc/src/main/java/org/factcast/server/grpc/auth/AccessRules.java [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+            return false;
+
+        if (include.stream().filter(s -> matches(s, ns)).findAny().isPresent())
+            return true;
+
+        return null;
+    }
+
+    private boolean matches(@NonNull String pattern, @NonNull String ns) {
+        if (pattern.equals(ns) || pattern.equals("*"))
```
Comment (uweschaefer) [inline factcast-server-grpc/src/main/java/org/factcast/server/grpc/auth/AccessRules.java]: Codacy Issue found: [Position literals first in String comparisons](https://app.codacy.com/manual/uwe/factcast/pullRequest?prid=5054049)

File: factcast-server-grpc/src/main/java/org/factcast/server/grpc/auth/AccessRules.java [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+            return true;
+
+        if (pattern.endsWith("*") && ns.startsWith(pattern.substring(0, pattern.length() - 1)))
```
Comment (uweschaefer) [inline factcast-server-grpc/src/main/java/org/factcast/server/grpc/auth/AccessRules.java]: Codacy Issue found: [Avoid unnecessary if..then..else statements when returning booleans](https://app.codacy.com/manual/uwe/factcast/pullRequest?prid=5054049)

**Comment by uweschaefer** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- factcast-server-grpc/src/main/java/org/factcast/server/grpc/auth/FactCastAccount.java  3
- factcast-server-grpc/src/main/java/org/factcast/server/grpc/auth/FactCastUser.java  1
- factcast-server-grpc/src/main/java/org/factcast/server/grpc/auth/FactCastAccessConfiguration.java  1
- factcast-server-grpc/src/main/java/org/factcast/server/grpc/auth/AccessRules.java  5
- factcast-server-grpc/src/main/java/org/factcast/server/grpc/FactCastSecurityConfiguration.java  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/uwe/factcast/pullRequest?prid=5054049&bid=16638135)

**Comment by uweschaefer** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- factcast-server-grpc/src/main/java/org/factcast/server/grpc/auth/FactCastAccessConfiguration.java  1
- factcast-server-grpc/src/main/java/org/factcast/server/grpc/FactCastSecurityConfiguration.java  1
- factcast-server-grpc/src/main/java/org/factcast/server/grpc/auth/AccessRules.java  5
- factcast-server-grpc/src/main/java/org/factcast/server/grpc/auth/FactCastUser.java  1
- factcast-server-grpc/src/main/java/org/factcast/server/grpc/auth/FactCastAccount.java  4
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/uwe/factcast/pullRequest?prid=5054049&bid=16638135)

**Comment by uweschaefer** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- factcast-server-grpc/src/main/java/org/factcast/server/grpc/auth/FactCastAccessConfiguration.java  1
- factcast-server-grpc/src/main/java/org/factcast/server/grpc/auth/FactCastAccount.java  4
- factcast-server-grpc/src/main/java/org/factcast/server/grpc/auth/FactCastUser.java  1
- factcast-server-grpc/src/main/java/org/factcast/server/grpc/auth/AccessRules.java  5
- factcast-server-grpc/src/main/java/org/factcast/server/grpc/FactCastSecurityConfiguration.java  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/uwe/factcast/pullRequest?prid=5054049&bid=16638135)

**Comment by uweschaefer** [comment]
closes #604

**Comment by uweschaefer** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- factcast-server-grpc/src/main/java/org/factcast/server/grpc/FactCastSecurityConfiguration.java  1
- factcast-server-grpc/src/main/java/org/factcast/server/grpc/auth/AccessRules.java  5
- factcast-server-grpc/src/main/java/org/factcast/server/grpc/auth/FactCastAccessConfiguration.java  1
- factcast-server-grpc/src/main/java/org/factcast/server/grpc/auth/FactCastAccount.java  4
- factcast-server-grpc/src/main/java/org/factcast/server/grpc/auth/FactCastUser.java  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/uwe/factcast/pullRequest?prid=5054049&bid=16638135)

**Comment by uweschaefer** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- factcast-server-grpc/src/main/java/org/factcast/server/grpc/auth/FactCastAccount.java  4
- factcast-server-grpc/src/main/java/org/factcast/server/grpc/auth/FactCastUser.java  1
- factcast-server-grpc/src/main/java/org/factcast/server/grpc/auth/FactCastAccessConfiguration.java  1
- factcast-server-grpc/src/main/java/org/factcast/server/grpc/auth/AccessRules.java  5
- factcast-server-grpc/src/main/java/org/factcast/server/grpc/FactCastSecurityConfiguration.java  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/uwe/factcast/pullRequest?prid=5054049&bid=16638135)

**Comment by uweschaefer** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- factcast-server-grpc/src/main/java/org/factcast/server/grpc/auth/AccessRules.java  5
- factcast-server-grpc/src/main/java/org/factcast/server/grpc/auth/FactCastAccessConfiguration.java  1
- factcast-server-grpc/src/main/java/org/factcast/server/grpc/auth/FactCastAccount.java  4
- factcast-server-grpc/src/main/java/org/factcast/server/grpc/auth/FactCastUser.java  1
- factcast-server-grpc/src/main/java/org/factcast/server/grpc/FactCastSecurityConfiguration.java  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/uwe/factcast/pullRequest?prid=5054049&bid=16638135)

**Comment by uweschaefer** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- factcast-server-grpc/src/main/java/org/factcast/server/grpc/auth/FactCastAccessConfiguration.java  1
- factcast-server-grpc/src/main/java/org/factcast/server/grpc/auth/FactCastAccount.java  4
- factcast-server-grpc/src/main/java/org/factcast/server/grpc/auth/FactCastUser.java  1
- factcast-server-grpc/src/main/java/org/factcast/server/grpc/auth/AccessRules.java  5
- factcast-server-grpc/src/main/java/org/factcast/server/grpc/FactCastSecurityConfiguration.java  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/uwe/factcast/pullRequest?prid=5054049&bid=16638135)

**Comment by uweschaefer** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- factcast-server-grpc/src/main/java/org/factcast/server/grpc/auth/FactCastAccount.java  4
- factcast-server-grpc/src/main/java/org/factcast/server/grpc/auth/FactCastAccessConfiguration.java  1
- factcast-server-grpc/src/main/java/org/factcast/server/grpc/auth/FactCastUser.java  1
- factcast-server-grpc/src/main/java/org/factcast/server/grpc/auth/AccessRules.java  5
- factcast-server-grpc/src/main/java/org/factcast/server/grpc/FactCastSecurityConfiguration.java  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/uwe/factcast/pullRequest?prid=5054049&bid=16638135)

**Comment by uweschaefer** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- factcast-server-grpc/src/main/java/org/factcast/server/grpc/FactCastSecurityConfiguration.java  1
- factcast-server-grpc/src/main/java/org/factcast/server/grpc/auth/FactCastUser.java  1
- factcast-server-grpc/src/main/java/org/factcast/server/grpc/auth/FactCastAccessConfiguration.java  1
- factcast-server-grpc/src/main/java/org/factcast/server/grpc/auth/FactCastAccount.java  4
- factcast-server-grpc/src/main/java/org/factcast/server/grpc/auth/AccessRules.java  5
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/uwe/factcast/pullRequest?prid=5054049&bid=16638135)

**Comment by uweschaefer** [comment]
see #677

**Comment by uweschaefer** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- factcast-server-grpc/src/main/java/org/factcast/server/grpc/auth/FactCastUser.java  1
- factcast-server-grpc/src/main/java/org/factcast/server/grpc/auth/AccessRules.java  5
- factcast-server-grpc/src/main/java/org/factcast/server/grpc/auth/FactCastSecretsConfiguration.java  1
- factcast-server-grpc/src/main/java/org/factcast/server/grpc/auth/FactCastAccount.java  4
- factcast-server-grpc/src/main/java/org/factcast/server/grpc/auth/FactCastAccessConfiguration.java  1
- factcast-server-grpc/src/main/java/org/factcast/server/grpc/FactCastSecurityConfiguration.java  2
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/uwe/factcast/pullRequest?prid=5054049&bid=16638135)

**Comment by uweschaefer** [comment]
closes #677

**Comment by uweschaefer** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- factcast-server-grpc/src/main/java/org/factcast/server/grpc/auth/AccessRules.java  5
- factcast-server-grpc/src/main/java/org/factcast/server/grpc/auth/FactCastAccessConfiguration.java  1
- factcast-server-grpc/src/main/java/org/factcast/server/grpc/FactCastSecurityConfiguration.java  2
- factcast-server-grpc/src/main/java/org/factcast/server/grpc/auth/FactCastUser.java  1
- factcast-server-grpc/src/main/java/org/factcast/server/grpc/auth/FactCastSecretsConfiguration.java  1
- factcast-server-grpc/src/main/java/org/factcast/server/grpc/auth/FactCastAccount.java  4
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/uwe/factcast/pullRequest?prid=5054049&bid=16638135)

**Comment by uweschaefer** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- factcast-server-grpc/src/main/java/org/factcast/server/grpc/auth/FactCastAccount.java  4
- factcast-server-grpc/src/main/java/org/factcast/server/grpc/auth/FactCastSecretsConfiguration.java  1
- factcast-server-grpc/src/main/java/org/factcast/server/grpc/auth/FactCastUser.java  1
- factcast-server-grpc/src/main/java/org/factcast/server/grpc/FactCastSecurityConfiguration.java  2
- factcast-server-grpc/src/main/java/org/factcast/server/grpc/auth/FactCastAccessConfiguration.java  1
- factcast-server-grpc/src/main/java/org/factcast/server/grpc/auth/AccessRules.java  5
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/uwe/factcast/pullRequest?prid=5054049&bid=16638135)

**Comment by uweschaefer** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- factcast-server-grpc/src/main/java/org/factcast/server/grpc/FactCastSecurityConfiguration.java  2
- factcast-server-grpc/src/main/java/org/factcast/server/grpc/auth/FactCastAccount.java  4
- factcast-server-grpc/src/main/java/org/factcast/server/grpc/auth/FactCastAccessConfiguration.java  1
- factcast-server-grpc/src/main/java/org/factcast/server/grpc/auth/FactCastSecretsConfiguration.java  1
- factcast-server-grpc/src/main/java/org/factcast/server/grpc/auth/AccessRules.java  5
- factcast-server-grpc/src/main/java/org/factcast/server/grpc/auth/FactCastUser.java  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/uwe/factcast/pullRequest?prid=5054049&bid=16638135)

**Comment by mweirauch** [comment]
Am I missing some bits and pieces or is non-programmatic (without custom code) security configuration (e.g. via env-vars or some other property-based solution) currently not possible?

**Comment by uweschaefer** [comment]
you can provide two json files...
https://github.com/factcast/factcast/blob/issue604/factcast-site/documentation/content/setup/examples/grpc-config-basicauth.md

**Comment by uweschaefer** [comment]
probably factcast-access.json being part of your server and factcast-secrets.json pulled into at startup-time from a safe(er) place (aws?), but that's up to you.

**Comment by mweirauch** [comment]
I got the point with the two files before, but I should have bee more precise as to what I mean with non-programmatic configuration. My understanding s currently is, that if you don't provide an in-classpath-solution, you have to roll your own `FactCastAccessConfig` and `FactCastSecretsConfiguration` beans.

From an end-user POV I am forced to derive my own FactCast-derivative (as in the whole product) because I can't provide external security configuration (e.g. by the means of environment variables or an external configuration file letting a lone a secrets store).

The proposal e.g. with a custom docker-layer is lacking a bit in the sense that:
* I need custom code to tell FC where the files are (because you don't really want to manipulate the jar, right?)
* making it a docker image layer means either:
  * you build in the target environment where is the only place where these secrets can be found
    * making the final image contain configuration (contradicts the external configuration factor)
    * ideally you only build the artifact once and configure it externally
  * building in a "build" environment where the production credentials shouldn't be found
    * making the final image contain configuration (contradicts the external configuration factor)
    * so an image per runtime environment

So apart from making it property-configurable (in a spring-stylish way), you could at least make it environment settable. Wouldn't a "here is the access-config-file-location" and "here is the secrets-file-location" setting be at least desirable? Then you could at least derive the configs in the docker-entrypoint, place it somewhere in the image *during exec* (not during build) and pass these config files along to the binary.

I am just trying to view it from an end-user perspective ... do I fork mongod, put in some code for reading my security configration or does mongod provide a means of telling it from where it shall read the configuration.

I might have thrown something together in the my review branch, but as my time is limited, no promises.

File: factcast-site/documentation/content/setup/examples/grpc-config-basicauth.md [hunk 3]
```diff
@@ -40,40 +40,103 @@ see module [examples/factcast-example-client-basicauth](https://github.com/factc
 
 On the server, in order to provide downward compatibility, security is disabled by default (this will change in the future, mongoDB taught us well). Once security is enabled, non-authenticated users will not be allowed to work with the grpc factstore anymore.
 
-There are three roles a client can take:
+In order to enable security, a Bean of type `FacCastAccessConfig` must be defined. This is done either by providing one in your FactCast Server's context, or by using the dead-simple approach to put a `factcast-access.json` on the root of your classpath to deserialize it from there.
```
Comment (mweirauch) [inline factcast-site/documentation/content/setup/examples/grpc-config-basicauth.md]: FacCastAccessConfig -> FactCastAccessConfig

File: factcast-server-grpc/src/main/java/org/factcast/server/grpc/FactCastSecurityConfiguration.java [hunk 4]
```diff
@@ -102,22 +163,31 @@ DaoAuthenticationProvider daoAuthenticationProvider(UserDetailsService uds) {
...
+            @org.springframework.beans.factory.annotation.Value("${insecure:#{false}}") Boolean insecureIsOk) {
+        if (insecureIsOk) {
+            log.warn(
+                    "**** FactCast Security is disabled. This is discouraged for production environments. You have been warned. ****");
+            return username -> new FactCastUser(FactCastAccount.GOD, "DISABLED");
+        }
+
+        log.error("**** FactCast Security is disabled. ****");
+        log.error("* If you really want to, you can run Factcast in such a configuration ");
+        log.error("* by adding a property '-Dinsecure' to your setup. However, it is");
```
Comment (mweirauch) [inline factcast-server-grpc/src/main/java/org/factcast/server/grpc/FactCastSecurityConfiguration.java]: `-Dinsecure` doesn't work due to it being the "big" boolean. It's `null` then. It needs an explicit value assigned.

**Comment by uweschaefer** [comment]
Hmm... you are not supposed to run FactCast from a provided jar, remember this is much like https://github.com/codecentric/spring-boot-admin, where you create your own to build, configure and package and use the server dependencies, so that the mongod example does not fit really. See examples/factcast-server*.

However, if we can make it easier, i am all for it.
Creating properties for the filelocations with classpath:/factcast-access.json etc as defaults if a good idea - will do.

**Comment by uweschaefer** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- factcast-server-grpc/src/main/java/org/factcast/server/grpc/FactCastSecurityConfiguration.java  2
- factcast-server-grpc/src/main/java/org/factcast/server/grpc/auth/FactCastSecretsConfiguration.java  1
- factcast-server-grpc/src/main/java/org/factcast/server/grpc/auth/FactCastAccessConfiguration.java  1
- factcast-server-grpc/src/main/java/org/factcast/server/grpc/auth/AccessRules.java  5
- factcast-server-grpc/src/main/java/org/factcast/server/grpc/auth/FactCastUser.java  1
- factcast-server-grpc/src/main/java/org/factcast/server/grpc/auth/FactCastAccount.java  4
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/uwe/factcast/pullRequest?prid=5054049&bid=16638135)

**Comment by uweschaefer** [comment]
@mweirauch would you suggest to provide the secrets as json via a single property?

**Comment by uweschaefer** [comment]
to make it simple, we just moved to single properties per user/secret. This leaves the door open for any PropertySource without any special handling.

**Comment by uweschaefer** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- factcast-server-grpc/src/main/java/org/factcast/server/grpc/FactCastSecretProperties.java  1
- factcast-server-grpc/src/main/java/org/factcast/server/grpc/auth/FactCastAccessConfiguration.java  1
- factcast-server-grpc/src/main/java/org/factcast/server/grpc/auth/FactCastUser.java  1
- factcast-server-grpc/src/main/java/org/factcast/server/grpc/FactCastSecurityConfiguration.java  5
- factcast-store-pgsql/src/main/java/org/factcast/store/pgsql/validation/ScheduledRegistryRefresher.java  1
- factcast-server-grpc/src/main/java/org/factcast/server/grpc/auth/FactCastAccount.java  4
- factcast-server-grpc/src/main/java/org/factcast/server/grpc/auth/AccessRules.java  5
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/uwe/factcast/pullRequest?prid=5054049&bid=16638135)

**Comment by uweschaefer** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- factcast-server-grpc/src/main/java/org/factcast/server/grpc/auth/FactCastAccessConfiguration.java  1
- factcast-server-grpc/src/main/java/org/factcast/server/grpc/auth/FactCastAccount.java  4
- factcast-store-pgsql/src/main/java/org/factcast/store/pgsql/validation/ScheduledRegistryRefresher.java  1
- factcast-server-grpc/src/main/java/org/factcast/server/grpc/FactCastSecretProperties.java  1
- factcast-server-grpc/src/main/java/org/factcast/server/grpc/auth/FactCastUser.java  1
- factcast-server-grpc/src/main/java/org/factcast/server/grpc/auth/AccessRules.java  5
- factcast-server-grpc/src/main/java/org/factcast/server/grpc/FactCastSecurityConfiguration.java  5
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/uwe/factcast/pullRequest?prid=5054049&bid=16638135)

File: factcast-site/documentation/content/migration/_index.md [hunk 5]
```diff
@@ -12,6 +12,17 @@ identifier = "migration"
...
+
+#### basic-auth setup has changed
+
+If you used a 'factcast-security.json' before, you will be please to learn that factcast was extended to support role/namespace based autorisation. Also the filename changed to 'factcast-access.json'.
+
+see [basicauth usage](/setup/examples/grpc-config-basicauth)
+
+#### basic-auth setup is enforced
+
+By default, when executing without security enabled, you need to supply a property 'insecure=true' via commandline or propertyfile to get away with just a warning. If you don't, factcast will exit with errorcode 1.
```
Comment (otbe) [inline factcast-site/documentation/content/migration/_index.md]: Just a question: why is it `insecure` instead of something like `factcast.insecure`?

**Comment by uweschaefer** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- factcast-server-grpc/src/main/java/org/factcast/server/grpc/FactCastSecretProperties.java  1
- factcast-store-pgsql/src/main/java/org/factcast/store/pgsql/validation/ScheduledRegistryRefresher.java  1
- factcast-server-grpc/src/main/java/org/factcast/server/grpc/auth/FactCastUser.java  1
- factcast-server-grpc/src/main/java/org/factcast/server/grpc/auth/AccessRules.java  5
- factcast-server-grpc/src/main/java/org/factcast/server/grpc/auth/FactCastAccount.java  4
- factcast-server-grpc/src/main/java/org/factcast/server/grpc/FactCastSecurityConfiguration.java  5
- factcast-server-grpc/src/main/java/org/factcast/server/grpc/auth/FactCastAccessConfiguration.java  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/uwe/factcast/pullRequest?prid=5054049&bid=16638135)

**Comment by mweirauch** [comment]
I don't have much time or rather no time this week to come up with a proposal. Started locally with `ConfigurationProperties` with a base prefix of `factcast.security`. Which by default would contain at least an `enabled` property because that is the common Spring nomenclature. So the property is scoped and not some lonely thingy un-namespaced.

If you could merge in the latest master changes I can try to provide this little proposal on my review branch.

Extending on this property based approach, one could also then provide these props via env vars or any other propery provisioning mechanism.

PS: Haven't had a look at the most recent changes.

**Comment by uweschaefer** [comment]
master is integrated. i will move the insecure attribute and we'll merge the branch due to the age of it. However, feel very welcome to open a new one.

**Comment by uweschaefer** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- factcast-store-pgsql/src/main/java/org/factcast/store/pgsql/validation/ScheduledRegistryRefresher.java  1
- factcast-server-grpc/src/main/java/org/factcast/server/grpc/auth/FactCastUser.java  1
- factcast-server-grpc/src/main/java/org/factcast/server/grpc/auth/FactCastAccount.java  4
- factcast-server-grpc/src/main/java/org/factcast/server/grpc/auth/FactCastAccessConfiguration.java  1
- factcast-server-grpc/src/main/java/org/factcast/server/grpc/auth/AccessRules.java  5
- factcast-server-grpc/src/main/java/org/factcast/server/grpc/FactCastSecretProperties.java  1
- factcast-server-grpc/src/main/java/org/factcast/server/grpc/FactCastSecurityConfiguration.java  5
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/uwe/factcast/pullRequest?prid=5054049&bid=16638135)

**Comment by uweschaefer** [comment]
we go for "factcast.security.enabled" defaulting to true

**Pull request merged**